src/modules/flow/iio/nodes.c

```diff
@@ -861,17 +873,26 @@ adc_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_opti
 }
 
 static bool
-light_create_channels(struct iio_double_data *mdata, int device_id)
+light_create_channels(struct iio_double_data *mdata, int device_id, const int channel_id)
 {
+    char channel_name[NAME_MAX];
+    int ret;
+
     mdata->iio_base.device = sol_iio_open(device_id, &mdata->iio_base.config);
     SOL_NULL_CHECK(mdata->iio_base.device, false);
 
-    mdata->channel_val = iio_add_channel(mdata->scale, mdata->offset, "in_illuminance0", &mdata->iio_base);
-    if (!mdata->channel_val) {
-        mdata->channel_val = iio_add_channel(mdata->scale, mdata->offset, "in_illuminance", &mdata->iio_base);
-        SOL_NULL_CHECK_GOTO(mdata->channel_val, error);
+    if (channel_id >= 0) {
+        ret = snprintf(channel_name, sizeof(channel_name), "in_illuminance%d", channel_id);
+    } else {
+        ret = snprintf(channel_name, sizeof(channel_name), "%s", "in_illuminance");
```

Comment:
Very minor: as this is an `snprintf` of `%s` only, a `strncpy(channel_name, "in_illumincance")` should suffice.